packages/cluster-ui/src/util/memoize.ts

```diff
@@ -0,0 +1,48 @@
+import * as protos from "@cockroachlabs/crdb-protobuf-client";
+import { longToInt } from "src/barCharts/utils";
+import { AggregateStatistics } from "src/statementsTable";
+
+type Statement = protos.cockroach.server.serverpb.StatementsResponse.ICollectedStatementStatistics;
+type statementType = AggregateStatistics | Statement;
+type statementsType = Array<statementType>;
+
+const cache = new Map();
+
+/**
+ * a simple hash function to minimaze the key size on the memoized function's cache
+ * @param s string to be hashed
+ * @returns the hashed string
+ */
+function stringToHash(s: string) {
+  let hash = 0;
+  if (s.length == 0) return hash;
+
+  for (let i = 0; i < s.length; i++) {
+    const char = s.charCodeAt(i);
+    hash = (hash << 5) - hash + char;
+    hash = hash & hash;
+  }
+  return hash;
+}
+
+/**
+ * Memoized function to calculate total workload of statements
+ * It will only recalculate if the list of statements change
+ * @param statements array of statements (AggregateStatistics or Statement)
+ * @returns the total workload of all statements
+ */
+export function memoizedTotalWorkload(statements: statementsType) {
+  const key = stringToHash(JSON.stringify(statements));
```

Comment:
I wonder if calculating the statement hash now becomes even more expensive than just recalculating the service latency because we're stringifying the statements object every time we want to get the statement hash. I think we should either:
1) Calculate this hash when first populating this statements data, so we can use a precomputed hash here (if the statements don't change, the hash won't either).
2) Forget about this for now. We might be overcomplicating this.

I'm happy to do 2. Over optimization is the root of all evil.

Reply:
made the changes to use a non memoized function

Reply:
Talking with @maryliag last night about this we were thinking through the same thing. It's the choice between either using the _stringified_ statements array as a key and risking the hit to memory or iterating over the array to get a unique value to cache the calculation all to save the cost of the calculation. I say we drop this and fix the performance problem when it arises. 

For posterity perhaps we can leave a nice descriptive comment about the performance and that there may be an opportunity to optimize in the future.